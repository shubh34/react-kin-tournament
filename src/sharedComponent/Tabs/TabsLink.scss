@import '../../colors';

.c-tab-button {
    width: 50%;
    background: $lightGray;
    color: white;
    font-weight: bold;
    &--active {
        background-color: $green;
        color: $darkGray;
    }
}