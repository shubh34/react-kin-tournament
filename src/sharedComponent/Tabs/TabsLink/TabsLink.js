import './TabsLink.scss';
import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import Button from '../../Button/Button';

const TabsLink = ({ name, onClick, isActive = false }) => {
	const className = classnames('c-tab-button',
		{
			'c-tab-button--active': isActive,
		});
	return (
		<Button type="button" id={name} name={name} className={className} onClick={onClick} />
	);
};


TabsLink.propTypes = {
	name: PropTypes.string.isRequired,
	onClick: PropTypes.func.isRequired,
	isActive: PropTypes.bool,
};

export default TabsLink;
