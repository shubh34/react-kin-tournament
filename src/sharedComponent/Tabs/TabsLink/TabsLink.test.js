import React from 'react';
import { shallow } from 'enzyme';
import TabsLink from './TabsLink';

describe('TabsLink', () => {
	it('renders without crashing', () => {
		const wrapper = shallow(<TabsLink name="TabLink" onClick={jest.fn()} />);
		expect(wrapper).toMatchSnapshot();
	});
	it('TabsLink when active', () => {
		const wrapper = shallow(<TabsLink name="TabLink" onClick={jest.fn()} isActive />);
		expect(wrapper).toMatchSnapshot();
	});
	it('should call onClick on Tablink click', () => {
		const onClick = jest.fn();
		const wrapper = shallow(<TabsLink name="TabLink" onClick={onClick} isActive />);
		wrapper.find('Button').simulate('click');
		expect(onClick).toBeCalled();
	});
});
