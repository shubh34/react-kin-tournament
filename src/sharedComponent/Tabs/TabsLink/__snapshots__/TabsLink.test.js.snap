// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsLink TabsLink when active 1`] = `
<Button
  className="c-tab-button c-tab-button--active"
  id="TabLink"
  name="TabLink"
  onClick={[MockFunction]}
  type="button"
/>
`;

exports[`TabsLink renders without crashing 1`] = `
<Button
  className="c-tab-button"
  id="TabLink"
  name="TabLink"
  onClick={[MockFunction]}
  type="button"
/>
`;
